src/services/codefixes/inferFromUsage.ts

```diff
@@ -201,15 +218,71 @@ namespace ts.codefix {
+        // find
+        const paramTags = createMap<JSDocParameterTag>();
+        forEach(parameterInferences, inference => {
             const param = inference.declaration;
             const typeNode = inference.type && getTypeNodeIfAccessible(inference.type, param, program, host);
-            return typeNode && !param.initializer && !getJSDocType(param) ? { ...inference, typeNode } : undefined;
+            if (typeNode && !param.initializer && !getJSDocType(param) && isIdentifier(inference.declaration.name)) {
+                paramTags.set(
+                    inference.declaration.name.escapedText as string,
+                    createJSDocParamTag(typeNode, inference.declaration.name as EntityName, !!inference.isOptional, ""));
+            }
         });
-        changes.tryInsertJSDocParameters(sourceFile, result);
+
+        // resolve
+        const signature = parameterInferences.length && parameterInferences[0].declaration.parent;
+        if (signature) {
+            insertMergedJSDoc(
+                paramTags, changes, sourceFile, signature,
+                t => isJSDocParameterTag(t) && isIdentifier(t.name) && t.name.escapedText as string,
+                (synthetic, existing) => {
+                    synthetic.comment = existing.comment;
+                    synthetic.isBracketed = existing.isBracketed;
+                });
+        }
+    }
+
+    /**
+     * @param getKey When getKey returns a string, it is also asserting that `t`, the existing tag whose key is returned,
+     * is of type T, where T is the type of the new, synthetic tags.
+     */
+    function insertMergedJSDoc<T extends JSDocTag>(
+        syntheticTags: Map<T>,
+        changes: textChanges.ChangeTracker,
+        sourceFile: SourceFile,
+        parent: HasJSDoc,
+        getKey: (tag: JSDocTag) => string | false,
+        merge: (synthetic: T, existing: T) => void) {
+        const comments = [];
+        const tags: JSDocTag[] = [];
+        if (hasJSDocNodes(parent)) {
+            for (const jsdoc of parent.jsDoc!) {
+                if (jsdoc.comment) {
+                    comments.push(jsdoc.comment);
+                }
+                if (jsdoc.tags) {
+                    for (const existing of jsdoc.tags) {
+                        const key = getKey(existing);
+                        const synthetic = key && syntheticTags.get(key);
+                        if (synthetic) {
+                            merge(synthetic, existing as T);
+                            tags.push(synthetic);
+                            syntheticTags.delete(key as string);
+                        }
+                        else {
+                            tags.push(existing);
+                        }
+                    }
+                }
+            }
+        }
+        syntheticTags.forEach(v => tags.push(v));
+        const tag = createJSDocComment(comments.join("\n"), createNodeArray(tags));

```

Comment:
>"\n" [](start = 53, length = 4)

Some later step will replace this with a platform-specific line break?

Reply:
Yes, the formatting code re-breaks lines and reindents them as well.